.github/workflows/test.yaml

```diff
@@ -0,0 +1,59 @@
+name: test
+
+on:
+  pull_request:
+
+jobs:
+  generate:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+          cache: true
+      - run: go get ./...
+      - run: go generate -x ./...
+      - uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.os }}-libraries
+          path: llm/llama.cpp/*/build/*/bin/*
```

Comment:
`llama.cpp/gguf/build/lib/*` is where the dynamic loaded libraries go.
`llama.cpp/gguf/ggml-metal.metal` is where we'll pull the metal file for darwin

The "built in" version for linux links against the archives from the CPU build variant located here:
`llm/llama.cpp/gguf/build/cpu/examples/server/libext_server.a`
`llm/llama.cpp/gguf/build/cpu/common/libcommon.a`
`llm/llama.cpp/gguf/build/cpu/libllama.a`
`llm/llama.cpp/gguf/build/cpu/libggml_static.a`

The built-in darwin is linked against these
`llm/llama.cpp/gguf/build/metal/common/libcommon.a`
`llm/llama.cpp/gguf/build/metal/examples/server/libext_server.a`
`llm/llama.cpp/gguf/build/metal/libllama.a`
`llm/llama.cpp/gguf/build/metal/libggml_static.a`


I can see this is a bit messy, so we may want to follow up with some post-processing step(s) in the generate script to aggregate these into a more consistent/clean location.

Reply:
Additional breadcrumbs:
* https://github.com/jmorganca/ollama/blob/main/llm/shim_darwin.go#L12
* https://github.com/jmorganca/ollama/blob/main/llm/shim_ext_server.go#L28
* https://github.com/jmorganca/ollama/blob/main/llm/ext_server.go#L13-L16
* https://github.com/jmorganca/ollama/blob/main/llm/ext_server.go#L20-L23